simapp/simd/cmd/root.go

```diff
@@ -39,8 +40,16 @@ import (
 
 // NewRootCmd creates a new root command for simd. It is called once in the
 // main function.
-func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
-	encodingConfig := simapp.MakeTestEncodingConfig()
+func NewRootCmd() *cobra.Command {
+	// we "pre"-instantiate the application for getting the injected/configured encoding configuration
+	tempApp := simapp.NewSimApp(log.NewNopLogger(), dbm.NewMemDB(), nil, true, simutil.EmptyAppOptions{})
```

Comment:
Actually, with the previous behavior we were never certain that the `InterfaceRegistry`, `AppCodec` or `TxConfig` were identical when using app wiring (as these were injected). This fixes it and as well ensures that we use the same params for the legacy app.